server_code/helpers.py

```diff
@@ -13,7 +13,10 @@ def hash_media(media):
     return hasher.hexdigest()
 
 
-def boolean_from_string(value, default):
+def boolean_from_string(value, default=False):
     options = {True: ["true", "yes", "1"], False: ["false", "no", "0"]}
-    options[default].append(None)
-    return value in options[default]
+    options[default].extend([None, ""])
+    str_to_bool = {v: _bool for _bool, _list in options.items() for v in _list}
+    if value is not None:
+        value = value.lower()
+    return str_to_bool.get(value, default)
```

Comment:
ahh looks like my comment didn't get posted

might be cleaner just to define the dict like this in the first instance
```
    str_to_bool = {
        "true": True, "yes": True, "1": True,
        "false": False, "no": False, "0": False,
        "": default, None: default
    }
```

Reply:
Yep. Definitely. One of those that's grown into a monster!